### PR TITLE
webbing rework!

### PR DESCRIPTION
**Pull request by ghost**
spider webs work differently now, here's the changes...

you can only place up to three levels in spider webs, so no infinite stacking spider webs

spider webs adhere to walls now, mostly just for show but they cannot breach a wall, say, to space without destroying the web on it first.

you can web tiles in an area around you instead of just the tile below

webs do not stop you from pulling if you're a spider - DONE

each level of spider web thickness has a different level of interrupting actions instead of every single spider web object running a prob to stop you from doing something - DONE

spider webs speed up spiders on them

refactors some bad code - I'M DOING IT

and god, some really nice sprites! - WIP

### CODE REVIEW

**Comment by Armhulen** [comment]
doesn't need to be open for testmerges

**Comment by Armhulen** [comment]
old pr: #37028

**Comment by optimumtact** [comment]
Please consider this a friendly warning, your PR's must have a properly descriptive title and body content that explains the purpose of the PR.

Your current title and or body content does not meet requirements, so please update it.

**Comment by ExcessiveUseOfCobblestone** [comment]
> testing new icons

what?

**Comment by Armhulen** [comment]
@optimumtact i'll get right to it, sorry. I was reusing an old pr

@ExcessiveUseOfCobblestone i was testing new cobbyweb sprites with onule, and it worked out pretty well

**Comment by ExcessiveUseOfCobblestone** [comment]
I believe the correct terminology is "pseudo-merging" 😜

**Comment by Armhulen** [comment]
get it? cobbywebs? :D?

**Comment by ExcessiveUseOfCobblestone** [comment]
💃💃💃💃💃💃💃💃💃💃💃💃💃💃💃💃💃💃💃💃💃💃💃💃💃💃💃💃💃💃
🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷🕷
🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸🕸

**Comment by Armhulen** [comment]
thanks for the support. donate to my patreon.

File: code/game/objects/effects/spiders.dm [hunk 1]
```diff
@@ -29,23 +29,56 @@
 
 /obj/structure/spider/stickyweb
 	icon_state = "stickyweb1"
+	var/max_integrity = 10
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/effects/spiders.dm]: this isn't a var already?

File: code/game/objects/effects/spiders.dm [hunk 2]
```diff
@@ -29,23 +29,56 @@
...
-		icon_state = "stickyweb2"
+	var/T = getTurf(src)
+	if(istype(T, turf/closed))
+		switch(rand(1,3))
+			if(1)
+				icon_state = "stickywall[level]-1"
+			if(2)
+				icon_state = "stickywall[level]-2"
+			if(3)
+				icon_state = "stickywall[level]-3"
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/effects/spiders.dm]: don't be a switch spammer

```
var/variant = rand(1,3)
icon_state = "stickywall[level]-[variant]"

File: code/game/objects/effects/spiders.dm [hunk 3]
```diff
@@ -29,23 +29,56 @@
...
+	else	
+		switch(rand(1,3))
+			if(1)
+				icon_state = "stickyweb[level]-1"
+			if(2)
+				icon_state = "stickyweb[level]-2"
+			if(3)
+				icon_state = "stickyweb[level]-3"
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/effects/spiders.dm]: see above but web instead

File: code/game/objects/effects/spiders.dm [hunk 4]
```diff
@@ -29,23 +29,56 @@
...
+	if(isliving(mover))
+		if(mover.pulledby)
+			if(istype(mover.pulledby, /mob/living/simple_animal/hostile/poison/giant_spider))
+				return 1
+		if(prob(mobwebbiness))
 			to_chat(mover, "<span class='danger'>You get stuck in \the [src] for a moment.</span>")
 			return 0
-	else if(istype(mover, /obj/item/projectile))
-		return prob(30)
 	return 1
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/effects/spiders.dm]: Go ahead and T/F these.

File: code/game/objects/effects/spiders.dm [hunk 5]
```diff
@@ -29,23 +29,56 @@
...
+				icon_state = "stickyweb[level]-3"
 	. = ..()
 
 /obj/structure/spider/stickyweb/CanPass(atom/movable/mover, turf/target)
 	if(istype(mover, /mob/living/simple_animal/hostile/poison/giant_spider))
 		return 1
-	else if(isliving(mover))
-		if(prob(50))
+	if(istype(mover, /obj/item/projectile))
+		return prob(projectilewebbiness)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/effects/spiders.dm]:
```
if(prob(projectilewebbiness)
    return TRUE
else
    return FALSE
```

File: code/game/objects/effects/spiders.dm [hunk 6]
```diff
@@ -29,23 +29,56 @@
 
 /obj/structure/spider/stickyweb
 	icon_state = "stickyweb1"
+	var/max_integrity = 10
+	var/mobwebbiness = 30 //prob that you get stuck in the web when moving through it
+	var/projectilewebbiness = 75 //same as above, but for projectiles.
+	var/level = 1 //used for picking sprites for the web
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/effects/spiders.dm]: Remove these numbers and put like "normal","heavy",etc. instead. 

spider1-1 spider2-2 vs spidernormal-1 spiderheavy-2

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/spiders.dm (hunk 6 above)
Comment (Armhulen) [reply code/game/objects/effects/spiders.dm]: green hill zone didn't die for these sins but OK FINE

**Comment by Armhulen** [comment]
ok sure this is only the lightest version of the webbing
unknown

**Comment by Armhulen** [comment]
you may now heart emoji

**Comment by Armhulen** [comment]
ggrrrrggrgrgr coming back to this i don't wanna let it stale out if i know i'm gonna be working on other things

**Pull request closed without merging**